src/test.cc

```diff
@@ -75,10 +75,12 @@ char* mkdtemp(char* name_template) {
 
 string GetSystemTempDir() {
 #ifdef _WIN32
-  char buf[1024];
-  if (!GetTempPath(sizeof(buf), buf))
+  WCHAR buf[MAX_PATH];
+  // GetTempPath returns zero if no path is found
+  if(GetTempPathW(MAX_PATH, buf) == 0)
     return "";
-  return buf;
+  string result = WideToUtf8(wstring(buf));
```

Comment:
You're constructing (aka, allocating and copying) a new wstring object from buffer.

Please modify the existing WideToUtf8 function (and the Utf8ToWide function), or make new versions of both, so that this is not necessary (Take a StringPiece, or take a CHAR_T* and size_t pair, either way)

Then call your new WideToUtf8 function as

```cpp
return WideToUtf8(buf);
```


While you're at it, if you also provide another parameter to your string conversion functions

```cpp
WideToUtf8(StringPiece data, size_t desiredLen = 0)
{
    lengthToAllocate = std::max(data.size(), desiredLen);
    ....
}
```

You also allow your conversion functions to be used efficiently with areas of the code where you know ahead of time that you'll be appending additional data. I pointed out one such place in one of my comments.